### PR TITLE
internal/function: use custom fixed struct as LOC result

### LINKED ISSUE #927: LOC function is returning external library object that might change
This query from the documentation:
```
SELECT
    LANGUAGE(file_path, blob_content) as lang,
    SUM(JSON_EXTRACT(LOC(file_path, blob_content), '$.Code')) as code,
    SUM(JSON_EXTRACT(LOC(file_path, blob_content), '$.Comments')) as comments,
    SUM(JSON_EXTRACT(LOC(file_path, blob_content), '$.Blanks')) as blanks,
    COUNT(1) as files
FROM commit_files
NATURAL JOIN refs
NATURAL JOIN blobs
WHERE ref_name='HEAD'
GROUP BY lang;
```
Actually is not working.

After updating goloc (a library that is used under the hood), they changed the output struct and instead of `Code` the output now is `code` and so on.

We should wrap that output struct to be sure that output JSON won't change if library changes.

### PR DESCRIPTION
**Pull request by erizocosmico**
Fixes #927 
 
 - [x] I updated the documentation explaining the new behavior if any.
 - [x] I updated CHANGELOG.md file adding the new feature or bug fix.
 - [x] I updated go-mysql-server using `make upgrade` command if applicable.
 - [x] I added or updated examples if applicable.
 - [x] I checked that changes on schema are reflected into the documentation, if applicable.

### CODE REVIEW

File: docs/using-gitbase/examples.md [hunk 1]
```diff
@@ -81,9 +81,9 @@ GROUP BY committer_email,
 ```sql
 SELECT
     LANGUAGE(file_path, blob_content) as lang,
-    SUM(JSON_EXTRACT(LOC(file_path, blob_content), '$.Code')) as code,
```
Comment (ajnavarro) [inline docs/using-gitbase/examples.md]: I would say to do the opposite, keep output as it was. Also, other functions have keys with first letter uppercase too.

Comment (erizocosmico) [reply docs/using-gitbase/examples.md]: Using uppercase is a bit weird in JSON, hence https://github.com/src-d/gitbase/issues/930

If we do want to use uppercase, though, I can close the issue and change it back here.

Comment (ajnavarro) [reply docs/using-gitbase/examples.md]: I'm concerned about backward compatibility. All queries that are using these functions will stop working with no error message...

**Comment by erizocosmico** [comment]
Struct tags are now uppercase. Docs added with the shape of LOC.

File: docs/using-gitbase/examples.md [hunk 2]
```diff
@@ -81,9 +81,9 @@ GROUP BY committer_email,
...
-    SUM(JSON_EXTRACT(LOC(file_path, blob_content), '$.Comments')) as comments,
```
Comment (ajnavarro) [inline docs/using-gitbase/examples.md]: Now you need to keep this example as it was.

File: docs/using-gitbase/functions.md [hunk 3]
```diff
@@ -17,7 +17,7 @@ To make some common tasks easier for the user, there are some functions to inter
 |`uast_xpath(blob, xpath) blob`| performs an XPath query over the given UAST nodes                                                                |
 |`uast_extract(blob, key) text array`| extracts information identified by the given key from the uast nodes                                       |
 |`uast_children(blob) blob`| returns a flattened array of the children UAST nodes from each one of the UAST nodes in the given array              |
-|`loc(path, blob) json`| returns a JSON map, containing the lines of code of a file, separated in three categories: Code, Blank and Comment lines |
+|`loc(path, blob) json`| returns a JSON map, containing the lines of code of a file, separated in three categories: code, blank and comment lines |
```
Comment (ajnavarro) [inline docs/using-gitbase/functions.md]: and this.

**Comment by erizocosmico** [comment]
Done

**Pull request merged**